File: library/src/main/java/com/qingmei2/library/di/scheduler/SchedulersModule.java
package com.qingmei2.library.di.scheduler;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by QingMei on 2017/11/17.
 * desc:
 */
@Module
public class SchedulersModule {

    public SchedulersModule() {
    }

    @Singleton
    @Provides
    SchedulerProvider providerSchedulers() {
        return new AppSchedulers();
    }
}

File: rhine/src/main/java/com/qingmei2/rhine/di/scheduler/SchedulersModule.kt
package com.qingmei2.rhine.di.scheduler

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Created by QingMei on 2017/11/17.
 * desc:
 */
@Module
class SchedulersModule {

    @Singleton
    @Provides
    internal fun providerSchedulers(): SchedulerProvider {
        return AppSchedulers()
    }
}
